doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,345 @@
+IPv6, and so on wherever it makes sense.
+
+## Vendor and technology independence
+
+Vendors and technologies are supported with zero bias. This means users can
+choose what's best for them, without being led by RIOT. Moreover, RIOT is Free
+Software [2], which means users are free to use it as they wish without
+lock-in.
+
+The choice of the hardware and standards RIOT supports should depend only on
+how well they support the use cases. Any allegiance that modules may have
+should be kept to themselves.
+
+## Versatility
+
+The use cases for IoT systems are monifold, and so are their requirements.
+Therefore, design decisions in RIOT should not prefer one technology or one
+protocol over another. 
+
+Default configurations should support as many users and use cases as possible.
+RIOT should provide everything users need, either in its code base or in the
+external packages it supports. The list of hardware, algorithms and other
+elements that RIOT supports should constantly be expanding.
+
+## Modularity
+
+RIOT decomposes into small modules. Being modular lets RIOT address many
+different memory, functionality, and performance demands. It also helps
+development efforts to scale in a widely distributed community.
+
+Modules should be abstracted from one another as cleanly as possible. It should
+be easy for users to manage or exploit this modularity, for example through
+configuration methods, easy integration of third-party source code, or
+different levels of modularity. The granularity of modules should be stable and
+similar across different types of module. Users shouldn't want to split
+modules, but they shouldn't be unnecessarily fine-grained either.
+
+## Cross-hardware portability
+
+Users might want to write a program for one piece of hardware, and later run it
+on another. RIOT should let user code be completely portable, so long as it's
+valid with the hardware.
+
+The hardware abstraction layer should be stable, well defined and consistent.
+Above the HAL, the only thing that modules should know about hardware is
+whether a required feature is there or not. If it isn't, the module should
+adapt accordingly, or not compile.
+
+## Real-time capabilities
+
+Different real-time guarantees are required for different use cases. Low
+frequency sensing needs only soft real-time support and can handle less timing
+accuracy so long as the timers support long timescales. High-frequency sensing
+and control applications need hard real-time guarantees and sub-millisecond
+timing accuracy.
+
+RIOT should deliver soft real-time performance by default, to cover the widest
+range of use cases. Programs with hard deadlines should be achievable with more
+expertise. It should provide timers which can competently handle the timescales
+of any application.
+
+## Interoperability
+
+RIOT nodes need to communicate reliably with non-RIOT nodes. The protocols need
+to be familiar to them and behave in the way they expect.
```

Comment:
This sounds confusing to me. Maybe replace by:
"RIOT nodes need to communicate reliably with non-RIOT nodes by carefully implementing open standards or identical technological specs."